definitions/npm/winston_v3.x.x/flow_v0.34.x-/winston_v3.x.x.js

```diff
@@ -1,100 +1,100 @@
+    exitOnError?: boolean,
+    format?: $winstonFormat,
+    level?: $Keys<T>,
+    levels?: T,
+    transports?: Array<$winstonTransport>
+  };
 
-declare type $winstonLogger<T: $winstonLevels> = {
-  [$Keys<T>]: (message: string, meta?: Object) => void,
-  add: $winstonTransport => void,
-  clear: () => void,
-  configure: ($winstonLoggerConfig<T>) => void,
-  log: (message: $winstonInfo<T>) => void,
-  remove: $winstonTransport => void
-};
+  declare export type $winstonLogger<T: $winstonLevels> = {
+    [$Keys<T>]: (message: string, meta?: Object) => void,
+    add: $winstonTransport => void,
+    clear: () => void,
+    configure: ($winstonLoggerConfig<T>) => void,
+    log: (message: $winstonInfo<T>) => void,
+    remove: $winstonTransport => void
+  };
 
-declare type $winstonConfigSubModule = {
-  npm: () => $winstonNpmLogLevels
-};
+  declare export type $winstonConfigSubModule = {
+    npm: () => $winstonNpmLogLevels
+  };
 
-declare type $winstonFormatSubModule = {
-  combine: (...args: Array<$winstonFormat>) => $winstonFormat,
-  json: () => $winstonFormat,
-  label: (config?: Object) => $winstonFormat,
-  metadata: () => $winstonFormat,
-  prettyPrint: () => $winstonFormat,
-  simple: () => $winstonFormat,
-  splat: () => $winstonFormat,
-  timestamp: (?{ alias?: string, format?: string }) => $winstonFormat,
-  colorize: () => $winstonFormat,
-  logstash: () => $winstonFormat,
-  printf: ((args: $winstonInfo<Object>) => string) => $winstonFormat
-};
+  declare export type $winstonFormatSubModule = {
+    ((info: Object) => Object): () => $winstonFormat,
+    combine: (...args: Array<$winstonFormat>) => $winstonFormat,
+    json: () => $winstonFormat,
+    label: (config?: Object) => $winstonFormat,
+    metadata: () => $winstonFormat,
+    prettyPrint: () => $winstonFormat,
+    simple: () => $winstonFormat,
+    splat: () => $winstonFormat,
+    timestamp: (?{ alias?: string, format?: string }) => $winstonFormat,
+    colorize: () => $winstonFormat,
+    logstash: () => $winstonFormat,
+    printf: ((args: $winstonInfo<Object>) => string) => $winstonFormat
+  };
 
-declare type $winstonDefaultLogger = $winstonLogger<$winstonNpmLogLevels>;
+  declare export type $winstonDefaultLogger = $winstonLogger<$winstonNpmLogLevels>;
 
-declare class $winstonContainer<T> {
-  constructor(config?: $winstonLoggerConfig<T>): $winstonContainer<T>;
-  add(loggerId: string, config?: $winstonLoggerConfig<T>): $winstonLogger<T>;
-  get(loggerId: string): $winstonLogger<T>;
-}
+  declare export class $winstonContainer<T> {
+    constructor(config?: $winstonLoggerConfig<T>): $winstonContainer<T>;
+    add(loggerId: string, config?: $winstonLoggerConfig<T>): $winstonLogger<T>;
+    get(loggerId: string): $winstonLogger<T>;
+    has(loggerId: string): boolean;
+  }
 
-declare module "winston" {
-  declare module.exports: $winstonDefaultLogger & {
+  declare export default $winstonDefaultLogger & {
```

Comment:
The problem is that this will break badly for any commonjs user. 
Also the lib is written using commonjs.

Reply:
Is there any way to export types in Common JS module?

Reply:
Did some quick testing and it's fine to use either `declare type` or `declare export type` in a commonjs module declaration, but that works only with a `type`.

So for classes one could use aliases:
```
declare class $WinstonTransport {}
declare export type WinstonTransport = typeof $WinstonTransport;
```
Unless there's a better solution.

Reply:
I made Common JS module with type exports as you suggested.